web/docs/auth/social-auth/keycloak.md

```diff
@@ -0,0 +1,547 @@
+        configFn: import { getConfig } from "@src/auth/keycloak.js",
+        // highlight-next-line
+        userSignupFields: import { userSignupFields } from "@src/auth/keycloak.js"
+      }
+    },
+    onAuthFailedRedirectTo: "/login"
+  },
+}
+
+entity User {=psl
+    id                        Int     @id @default(autoincrement())
+    username                  String  @unique
+    displayName               String
+psl=}
+
+// ...
+```
+
+```js title=src/auth/keycloak.js
+export const userSignupFields = {
+  username: () => "hardcoded-username",
+  displayName: (data) => data.profile.name,
+}
+
+export function getConfig() {
+  return {
+    scopes: ['profile', 'email'],
+  }
+}
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp title="main.wasp"
+app myApp {
+  wasp: {
+    version: "^0.13.0"
+  },
+  title: "My App",
+  auth: {
+    userEntity: User,
+    methods: {
+      keycloak: {
+        // highlight-next-line
+        configFn: import { getConfig } from "@src/auth/keycloak.js",
+        // highlight-next-line
+        userSignupFields: import { userSignupFields } from "@src/auth/keycloak.js"
+      }
+    },
+    onAuthFailedRedirectTo: "/login"
+  },
+}
+
+entity User {=psl
+    id                        Int     @id @default(autoincrement())
+    username                  String  @unique
+    displayName               String
+psl=}
+
+// ...
+```
+
+```ts title=src/auth/keycloak.ts
+import { defineUserSignupFields } from 'wasp/server/auth'
+
+export const userSignupFields = defineUserSignupFields({
+  username: () => "hardcoded-username",
+  displayName: (data: any) => data.profile.name,
+})
+
+export function getConfig() {
+  return {
+    scopes: ['profile', 'email'],
+  }
+}
+```
+
+<GetUserFieldsType />
+
+</TabItem>
+</Tabs>
+
+## Using Auth
+
+<UsingAuthNote />
+
+## API Reference
+
+<ApiReferenceIntro />
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp title="main.wasp"
+app myApp {
+  wasp: {
+    version: "^0.13.0"
+  },
+  title: "My App",
+  auth: {
+    userEntity: User,
+    methods: {
+      keycloak: {
+        // highlight-next-line
+        configFn: import { getConfig } from "@src/auth/keycloak.js",
+        // highlight-next-line
+        userSignupFields: import { userSignupFields } from "@src/auth/keycloak.js"
+      }
+    },
+    onAuthFailedRedirectTo: "/login"
+  },
+}
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp title="main.wasp"
+app myApp {
+  wasp: {
+    version: "^0.13.0"
+  },
+  title: "My App",
+  auth: {
+    userEntity: User,
+    methods: {
+      keycloak: {
+        // highlight-next-line
+        configFn: import { getConfig } from "@src/auth/keycloak.js",
+        // highlight-next-line
+        userSignupFields: import { userSignupFields } from "@src/auth/keycloak.js"
+      }
+    },
+    onAuthFailedRedirectTo: "/login"
+  },
+}
+```
+
+</TabItem>
+</Tabs>
+
+The `keycloak` dict has the following properties:
```

Comment:
```suggestion
The `keycloak` dictionary has the following properties:
```

Reply:
We call the type in Wasp language `dict` https://wasp-lang.dev/docs/general/language#complete-list-of-wasp-types 🤷

Reply:
Ha, didn't know that!

I always thought the author was lazy whenever I read `dict`.